jwql/instrument_monitors/common_monitors/cosmic_ray_monitor.py

```diff
@@ -0,0 +1,608 @@
+
+                # Get observation time info
+                obs_start_time = jump_head['EXPSTART']
+                obs_end_time = jump_head['EXPEND']
+                start_time = julian.from_jd(obs_start_time, fmt='mjd')
+                end_time = julian.from_jd(obs_end_time, fmt='mjd')
+
+                cosmic_ray_num = len(jump_locs)
+                cosmic_ray_mags = self.get_cr_mags(jump_locs, jump_locs_pre, rate_data, jump_data, jump_head)
+
+                # Insert new data into database
+                try:
+                    cosmic_ray_db_entry = {'entry_date': datetime.datetime.now(),
+                                           'aperture': self.aperture,
+                                           'source_file': file_name,
+                                           'obs_start_time': start_time,
+                                           'obs_end_time': end_time,
+                                           'jump_count': cosmic_ray_num,
+                                           'magnitude': cosmic_ray_mags
+                                           }
+                    self.stats_table.__table__.insert().execute(cosmic_ray_db_entry)
+
+                    logging.info("Successfully inserted into database. \n")
+                except:
+                    logging.info("Could not insert entry into database. \n")
+
+    @log_fail
+    @log_info
+    def run(self):
+        """The main method. See module docstrings for additional info
+
+        Queries MAST for new MIRI data and copies it to a working
+        directory where it is run through the JWST pipeline. The output
+        of the 'jump' and 'rate' steps is used to determine the number
+        and magnitudes of cosmic rays which is then saved to the
+        database.
+        """
+
+        logging.info('Begin logging for cosmic_ray_monitor')
+
+        self.query_end = Time.now().mjd
+
+        for instrument in JWST_INSTRUMENT_NAMES:
+            if instrument == 'miri' or instrument == 'nircam':
+                self.instrument = instrument
+
+                # Identify which tables to use
+                self.identify_tables()
+
+                # Get a list of possible apertures
+                #possible_apertures = list(Siaf(instrument).apernames)
+                possible_apertures = self.possible_apers(instrument)
+
+                # Use this line instead to save time while testing
+                #possible_apertures = ['MIRIM_FULL', 'NRCB4_FULL']
+
+                for aperture in possible_apertures:
+
+                    logging.info('')
+                    logging.info('Working on aperture {} in {}'.format(aperture, instrument))
+
+                    self.aperture = aperture
+
+                    # We start by querying MAST for new data
+                    self.query_start = self.most_recent_search()
+
+                    logging.info('\tMost recent query: {}'.format(self.query_start))
+
+                    new_entries = self.query_mast()
```

Comment:
I think this search needs to be refined. I'm running this on the test server and it seems to be looking for all kinds of level 2 and level 3 products. This doesn't make sense when it is then going to search these for CRs, which have been removed by the end of stage 1. I think the best bet would be to limit the search to uncal.fits files, and then run the pipeline up through the jump step.

Reply:
Filtering using something like this might work:
https://github.com/spacetelescope/jwql/issues/935
Filter down to just unique file base names, then look for only the uncal and rate files.

Reply:
@bhilbert4 are we going to implement these changes?

Reply:
This has been implemented, in filter_bases()